tests/AwsClientTest.php

```diff
@@ -982,4 +985,19 @@ public function testClientParameterOverridesDefaultAccountIdEndpointModeBuiltIns
 
         self::assertEquals($expectedAccountIdEndpointMode, $builtIns['AWS::Auth::AccountIdEndpointMode']);
     }
+
+    public function testAppendsUserAgentMiddleware()
+    {
+        $client = new S3Client([
+            'region' => 'us-east-2',
+            'http_handler' => function (RequestInterface $request) {
+                $userAgentValue = $request->getHeaderLine('User-Agent');
```

Comment:
nit: I would add an extra assertion that confirms the header exists before asserting its value

Reply:
This is already handled by the method `getHeaderLine` itself. If the header is not present then it will return an empty string.